build failure on src/cql2ccl.c with C23 with bool usage

developing gcc15 defaults to C23, and yaz fails to compile with gcc15 (or `gcc14 -std=c23`) like:

```
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/usr/include/libxml2 -DWITH_GZFILEOP -I/usr/include/p11-kit-1 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -I/usr/include/hiredis -D_FILE_OFFSET_BITS=64 -c cql2ccl.c  -fPIC -DPIC -o .libs/cql2ccl.o
make[2]: Leaving directory '/builddir/build/BUILD/yaz-5.34.3-build/yaz-5.34.3/src'
cql2ccl.c:162:12: error: 'bool' cannot be used here
  162 | static int bool(struct cql_node *cn,
      |            ^~~~
cql2ccl.c:162:12: note: 'bool' is a keyword with '-std=c23' onwards
cql2ccl.c:162:17: error: expected identifier or '(' before 'struct'
  162 | static int bool(struct cql_node *cn,
      |                 ^~~~~~
cql2ccl.c: In function 'cql_to_ccl_r':
cql2ccl.c:242:16: error: expected expression before 'bool'
  242 |         return bool(cn, pr, client_data);
      |                ^~~~
make[2]: *** [Makefile:1120: cql2ccl.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Entering directory '/builddir/build/BUILD/yaz-5.34.3-build/yaz-5.34.3/src'
```

`bool` is keyword on C23.

--- a/src/cql2ccl.c
+++ b/src/cql2ccl.c
@@ -83,7 +83,7 @@ static void pr_term(const char **cpp, int stop_at_space,
     *cpp = cp;
 }
 
-static int node(struct cql_node *cn,
+static int node_st(struct cql_node *cn,
                 void (*pr)(const char *buf, void *client_data),
                 void *client_data)
 {
@@ -159,7 +159,7 @@ static int node(struct cql_node *cn,
 }
 
 
-static int bool(struct cql_node *cn,
+static int node_bool(struct cql_node *cn,
                 void (*pr)(const char *buf, void *client_data),
                 void *client_data)
 {
@@ -237,9 +237,9 @@ static int cql_to_ccl_r(struct cql_node *cn,
     switch (cn->which)
     {
     case CQL_NODE_ST:
-        return node(cn, pr, client_data);
+        return node_st(cn, pr, client_data);
     case CQL_NODE_BOOL:
-        return bool(cn, pr, client_data);
+        return node_bool(cn, pr, client_data);
     case CQL_NODE_SORT:
         return cql_to_ccl_r(cn->u.sort.search, pr, client_data);
     }

